pkg/florist/user.go

```diff
@@ -67,22 +68,41 @@ func SupplementaryGroups(username string, groups ...string) error {
 	return nil
 }
 
+type UserAddOpt struct {
+	UID int
+}
+
 // UserSystemAdd adds the system user 'username' and group 'username', with
 // home directory 'homedir' and mode 0o755.
-func UserSystemAdd(username string, homedir string) error {
+func UserSystemAdd(username string, homedir string, opts ...UserAddOpt) error {
```

Comment:
why `opts ...UserAddOpt`? It looks to me a single UserOpt struct would be enough?

This list adds unnecessary complexity and requires checks like: `optsCount != 1` and crazy invocations like: `opts[0].UID`. 

I don't see any value if having this as a list... If one wants more options you add more fields to existing stuct...
```
type UserAddOpt struct {
	UID int
    Option2 string
    Option3 bool
}
```

Reply:
and now I think that this is what is the less worse to do: breaking backwards compatibility and using as function signature
```
func userAdd(username string, opts UserAddOpt)
```

Reply:
I would like to think a bit about what to do about `UserAdd(username string, opts UserAddOpt)`.
At work I think that the simplest is to keep referring to your branch in #3 (eventually rebased after the merge of PrivateIP()). Or, at your discretion, copy the code in #3 somewhere in the work repo.
And please ping me again if I forget to take a decision :-)
merci!